### ISSUE TITLE
Two python virtualenv segments are shown

### ISSUE THREAD

**Issue by Zeta611** [romkatv/powerlevel10k#481]
Hi, I'm using p10k using the follwoing configuration wizard options:
```sh
# Wizard options: nerdfont-complete + powerline, small icons, rainbow, time,
# angled separators, sharp heads, flat tails, 2 lines, disconnected, no frame, sparse,
# many icons, fluent, transient_prompt, instant_prompt=verbose.
```
I have a pyenv-managed python (3.8.1) and a virtualenv created using it (named neovim-python3).
If I activate it, I see duplicate segments as below:
<img width="921" alt="image" src="https://user-images.githubusercontent.com/9553691/74102233-d8020a00-4b84-11ea-9146-52cc038b2485.png">

Is this an expected behavior, or can it be fixed?

I'm using:
zsh 5.7.1 (x86_64-apple-darwin19.0)
macOS Catalina (10.15.3)

**Comment by romkatv** [comment]
Is it possible that you also have `asdf` installed and have `legacy_version_file = yes` in `~/.asdfrc`?

**Comment by Zeta611** [comment]
No, I do not have `asdf` installed. But if it is relevant, I do have `pyenv`, `pyenv-virtualenv`, and `pipenv`.

**Comment by romkatv** [comment]
Let's try to figure out which two segments are displaying the same information. Try hiding the following segments one by one and see when your prompt changes.

- asdf
- virtualenv
- anaconda
- pyenv

To hide segment `foo`, type the following command:

```zsh
p10k display '*/foo'=hide
```

**Comment by Zeta611** [comment]
Thank you for a fast reply! I found out that the two were from 'pyenv' and 'virtualenv'. Since my `neovim-python3` is using `3.8.1` python, is there a way to change the pyenv prompt to show `3.8.1`, instead of the redundant `neovim-python3`

**Comment by romkatv** [comment]
What's the output of `pyenv version-name` when your prompt looks like on the screenshot?

**Comment by Zeta611** [comment]
It outputs `neovim-python3`.

**Comment by Zeta611** [comment]
This is the output of `pyenv versions`:
```sh
❯ pyenv versions
  system
  2.7.17
  2.7.17/envs/neovim-python2
  3.8.1
  3.8.1/envs/neovim-python3
  3.8.1/envs/tex-minted
  neovim-python2
* neovim-python3 (set by PYENV_VERSION environment variable)
  tex-minted
```

**Comment by romkatv** [comment]
`pyenv` prompt segment is supposed to always display the same thing as `pyenv version-name`, so it's working as intended.

Try the following configuration options. Do they make your prompt behave the way you want?

```zsh
# Display the output of `python --version` next to the virtual environment name.
typeset -g POWERLEVEL9K_VIRTUALENV_SHOW_PYTHON_VERSION=true
typeset -g POWERLEVEL9K_VIRTUALENV_{LEFT,RIGHT}_DELIMITER=

# Hide `pyenv` segment if its content has no dots ('.').
typeset -g POWERLEVEL9K_PYENV_CONTENT_EXPANSION='${(M)P9K_CONTENT:#*.*}'
typeset -g POWERLEVEL9K_PYENV_VISUAL_IDENTIFIER_EXPANSION='${${(M)P9K_CONTENT:#*.*}:+${P9K_VISUAL_IDENTIFIER// }}'
```

The code to hide `pyenv` is a bit obscure but hopefully at least the comment is clear. Basically, you'll never see `pyenv` segment if its content (a.k.a. text) has no dots in it.

**Comment by Zeta611** [comment]
<img width="830" alt="스크린샷 2020-02-09 오후 10 53 26" src="https://user-images.githubusercontent.com/9553691/74103382-fd941100-4b8e-11ea-9f06-0b1504a786b2.png">
This is exactly what I wanted! Thank you :tada:

**Comment by romkatv** [comment]
Great!

By the way, you can remove `vi_mode` from `POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS`. Prompt symbol displays the same information. It changes depending on vi mode: `❯`, `❮`, `Ⅴ`, `▶` for insert, command, visual and replace mode respectively.

**Comment by Zeta611** [comment]
Also good to know! :)

**Comment by tschmel** [comment]
Hi @romkatv ,
I was facing the same issue but your solution helped me a lot, thank you for this!
However, the python version which is displayed is not correct if you enable an environment via `pyenv activate` (see bottom right of the image attached). Can you also help me with this issue?

<img width="1119" alt="Bildschirmfoto 2020-05-15 um 15 37 58" src="https://user-images.githubusercontent.com/56080255/82056311-10ecd600-96c2-11ea-83c5-783c5a6102bd.png">

**Comment by romkatv** [comment]
@tschmel Please try removing `virtualenv` from `POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS` and adding this parameter:

```zsh
typeset -g POWERLEVEL9K_PYENV_CONTENT_EXPANSION='${P9K_CONTENT}${${P9K_PYENV_PYTHON_VERSION:#$P9K_CONTENT}:+ $P9K_PYENV_PYTHON_VERSION}'
```

Does this work the way you want?

**Comment by tschmel** [comment]
Thanks for your fast response, it works like a charm! 👌 

For future reference, this is what I had to add to my _.zshrc_ to get it running:

```zsh
# Define right side elements (all besides virtualenvs)
typeset -g POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS=(status command_execution_time background_jobs direnv asdf
    anaconda pyenv goenv nodenv nvm nodeenv rbenv rvm fvm luaenv jenv plenv phpenv haskell_stack kubecontext
    terraform aws aws_eb_env azure gcloud google_app_cred context nordvpn ranger nnn vim_shell midnight_commander
    nix_shell vi_mode todo timewarrior taskwarrior time)

# Display the output of `python --version` next to the virtual environment name.
typeset -g POWERLEVEL9K_VIRTUALENV_SHOW_PYTHON_VERSION=true
typeset -g POWERLEVEL9K_VIRTUALENV_{LEFT,RIGHT}_DELIMITER=

# Hide `pyenv` segment if its content has no dots ('.').
typeset -g POWERLEVEL9K_PYENV_CONTENT_EXPANSION='${(M)P9K_CONTENT:#*.*}'
typeset -g POWERLEVEL9K_PYENV_VISUAL_IDENTIFIER_EXPANSION='${${(M)P9K_CONTENT:#*.*}:+${P9K_VISUAL_IDENTIFIER// }}'

# Fix python version for pyenv virtualenvs
typeset -g POWERLEVEL9K_PYENV_CONTENT_EXPANSION='${P9K_CONTENT}${${P9K_PYENV_PYTHON_VERSION:#$P9K_CONTENT}:+ $P9K_PYENV_PYTHON_VERSION}'

```

**Comment by romkatv** [comment]
> ```shell
> # Display the output of `python --version` next to the virtual environment name.
> typeset -g POWERLEVEL9K_VIRTUALENV_SHOW_PYTHON_VERSION=true
> typeset -g POWERLEVEL9K_VIRTUALENV_{LEFT,RIGHT}_DELIMITER=
> 
> # Hide `pyenv` segment if its content has no dots ('.').
> typeset -g POWERLEVEL9K_PYENV_CONTENT_EXPANSION='${(M)P9K_CONTENT:#*.*}'
> typeset -g POWERLEVEL9K_PYENV_VISUAL_IDENTIFIER_EXPANSION='${${(M)P9K_CONTENT:#*.*}:+${P9K_VISUAL_IDENTIFIER// }}'
> ```

You need to delete these. My previous comment describes all changes that are necessary on top of a standard `~/.p10k.zsh`.

1. Remove `virtualenv` from `POWERLEVEL9K_RIGHT_PROMPT_ELEMENTS`
2. Add this parameter:
```zsh
typeset -g PWERLEVEL9K_PYENV_CONTENT_EXPANSION='${P9K_CONTENT}${${P9K_PYENV_PYTHON_VERSION:#$P9K_CONTENT}:+ $P9K_PYENV_PYTHON_VERSION}'
```

**Comment by tschmel** [comment]
You are absolutely right, I edited my previous comment accordingly.

**Comment by romkatv** [comment]
By the way, this comment isn't accurate:

```zsh
# Fix python version for pyenv virtualenvs
```

Parameter `PWERLEVEL9K_PYENV_CONTENT_EXPANSION` isn't fixing Python version. It defines the format of `pyenv` segment. Here's the login of that parameter's value in plain English.

- If `$P9K_PYENV_PYTHON_VERSION` is not empty and unequal to `$P9K_CONTENT`, display `$P9K_CONTENT $P9K_PYENV_PYTHON_VERSION`.
- Otherwise display just `$P9K_CONTENT`.

(`$P9K_CONTENT` is what you normally see without `CONTENT_EXPANSION`.)

**Comment by tschmel** [comment]
Thanks for the clarification. What about the following:

```
# Display pyenv and virtuelenv information in the expected format
```

**Comment by romkatv** [comment]
My goal is for you to understand what the parameter does. If you do, comment is up to you as it goes into your config.

**Comment by jvcarli** [comment]
Hi, I'm facing a similar doubt, but with ASDF and python venvs. When I activate my venv, ASDF is still shown. Can ASDF be hidden when venvs are activated?

![Screen Shot 2020-11-14 at 09 33 45](https://user-images.githubusercontent.com/57324218/99147074-824f1680-265c-11eb-8a1a-6d8ee077961e.png)

Thank you for powerlevel, it's amazing!

**Comment by romkatv** [comment]
@jvcarli Add this to `~/.p10k.zsh`:

```zsh
p10k-on-pre-prompt() {
  if [[ -n ${VIRTUAL_ENV-} ]]; then
    p10k display '*/asdf'=hide
  else
    p10k display '*/asdf'=show
  fi
}
```

The restart zsh with `exec zsh`.

In the future please open a separate issue instead of commenting on a closed one.
